### PR TITLE
fix: allow `torch.Symint` in `slice_scatter_decomposition`

### PR DESCRIPTION
**Pull request by jiwoong-choi**
fix: allow `torch.Symint` objects for `start`, `end` and `step` in `slice_scatter_decomposition`

# Description

When an exported program is `torch.export`ed with dynamic shapes, the `start`, `end` and `step` could be `torch.SymInt` instead of `int`. As a result, the assertions on `start`, `end` and `step` in `slice_scatter_decomposition` aborts the lowering pass.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [v] My code follows the style guidelines of this project (You can use the linters)
- [v] I have performed a self-review of my own code
- [v] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [v] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by peri044 — COMMENTED** [review]
I don't think our code can handle (start, end, step) to be `SymInt`. Can you provide a test case where you see such behavior ?

**Comment by jiwoong-choi** [comment]
I'll reopen this PR (or might open a new one with more changes) after I come with a proper test case.
